qiita_plugins/target_gene_type/target_gene_type/validate.py

```diff
@@ -0,0 +1,469 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from os.path import basename, join, splitext
+from json import loads
+from shutil import copy
+
+from h5py import File
+from qiita_client import format_payload
+from skbio.parse.sequences import load
+
+# This is temporary. We should move the demux file format somewhere else
+# (Jose) I'm planning to do this soon
+from qiita_ware.demux import to_hdf5, to_ascii, format_fasta_record
+
+FILEPATH_TYPE_DICT = {
+    'SFF': ({'raw_sff'}, set()),
+    'FASTQ': ({'raw_forward_seqs', 'raw_barcodes'}, {'raw_reverse_seqs'}),
+    'FASTA': ({'raw_fasta', 'raw_qual'}, set()),
+    'FASTA_Sanger': ({'raw_fasta'}, set()),
+}
+
+
+def _validate_multiple(qclient, job_id, prep_info, files, atype):
+    """Validae and fix a new 'SFF', 'FASTQ', 'FASTA' or 'FASTA_Sanger' artifact
```

Comment:
Should it just be Validate and fix a new a new FILEPATH_TYPE_DICT, see missing t ...